src/scripts/hooks/canvas-ready.ts

```diff
@@ -23,14 +23,24 @@ export const CanvasReady = {
                 CONFIG.MeasuredTemplate.defaults.angle = canvas.scene.hasHexGrid ? 60 : 90;
             }
 
-            Promise.resolve().then(async () => {
-                // Redraw tokens
-                for (const token of canvas.tokens.placeables) {
+            // Redraw tokens
+            for (const token of canvas.tokens.placeables) {
+                let redrawTrigger = Promise.resolve();
+
+                // Animated tokens sometimes disappear if the underlying video isn't given a moment to begin playing
+                if (token.isVideo) {
+                    redrawTrigger = new Promise((resolve) => {
+                        // Wait until 50ms after the video starts playing to be safe - sometimes they'll still be
+                        // invisible if drawn immediately after they begin playing
+                        token.sourceElement.addEventListener("play", () => setTimeout(resolve, 50));
```

Comment:
This is going to continuously add event listeners to the token.

Reply:
This is been in place in one form or another since V9, so maybe it's not needed anymore. Test the cases I mentioned before without the redraw present. If they still work, we can remove it entirely.

Reply:
> This is going to continuously add event listeners to the token.

I'm not quite sure what you mean - `new Promise` only calls the callback once, so only one event listener is added. The event listener should only ever fire once since the user cannot control the video element and it loops continuously, and `setTimeout` also fires only once.

I'll configure the listener with `{ once: true }` just to be safe, in case there's an edge case I'm missing with the video playback.

> This is been in place in one form or another since V9, so maybe it's not needed anymore. Test the cases I mentioned before without the redraw present. If they still work, we can remove it entirely.

Things seem to mostly work. The only issue I noticed is that if the enlarge effect is applied by a different user while you're viewing a different scene, switching back to the original scene sometimes shows tokens with the wrong size until you hover over/select a token. This seems like a minor enough bug that it might be fine to just remove this code anyways in the interest of simplifying things.